lib/datastore/request.js

```diff
@@ -617,11 +616,15 @@ DatastoreRequest.prototype.runQuery = function(query, callback) {
       return;
     }
 
-    var entities = entity.formatArray(resp.batch.entity_result);
+    var entities = [];
     var nextQuery = null;
 
-    if (resp.batch.end_cursor && entities.length > 0) {
-      var endCursor = resp.batch.end_cursor.toBase64();
+    if (resp.batch.entityResults) {
+      entities = entity.formatArray(resp.batch.entityResults);
+    }
+
+    if (resp.batch.endCursor && entities.length > 0) {
+      var endCursor = resp.batch.endCursor.toBase64();
 
       nextQuery = extend(true, new Query(), query);
       nextQuery.start(endCursor).offset(0);
```

Comment:
I think I understand, but if we set an offset on the `end_cursor` returned, wouldn't it offset from there?

Here's how I'm envisioning the API working:
```
Query:
Get the last 8 things (offset = 2)

10 things exist:
1
2
3 - match! - batch.results.push()
4 - match! - batch.results.push()
5 - match! - batch.results.push()
6 - I need a break! - batch.end_cursor = pointer to this result
7
8
9
10

Response:
{
  "results": [
    3,
    4,
    5
  ],
  end_cursor: "pointer to 6",
  skipped_results: 2,
  more_results: "NOT_FINISHED"
}
```

If we set the next query's offset to anything other than 0, wouldn't it start from "6" + offset, skipping over some of the results that matched the original query?